ast/dml.go

```diff
@@ -135,7 +135,29 @@ type TableName struct {
 
 // Restore implements Recoverable interface.
 func (n *TableName) Restore(sb *strings.Builder) error {
-	return errors.New("Not implemented")
+	//Joining indexHints.
+	indexHints := func(sb *strings.Builder) error {
+		for i, value := range n.IndexHints {
+			if i >= 0 {
+				sb.WriteString(" ")
+			}
+			if err := value.Restore(sb); err != nil {
+				return errors.Annotate(err, "An error occurred while splicing IndexHints")
+			}
+		}
+		return nil
+	}
+
+	if n.Schema.String() != "" {
+		WriteName(sb, n.Schema.String())
+		sb.WriteString(".")
+	}
+	WriteName(sb, n.Name.String())
+	if err := indexHints(sb); err != nil {
```

Comment:
Since `indexHints` is only used once here, perhaps just inline that function directly?